File: src/main/java/com/jss/osiris/modules/osiris/accounting/service/PaySlipLineTypeService.java
package com.jss.osiris.modules.osiris.accounting.service;

import java.util.List;

import com.jss.osiris.modules.osiris.accounting.model.PaySlipLineType;

public interface PaySlipLineTypeService {
    public List<PaySlipLineType> getPaySlipLineTypes();

    public PaySlipLineType getPaySlipLineType(Integer id);

    public PaySlipLineType addOrUpdatePaySlipLineType(PaySlipLineType paySlipLineType);

    public PaySlipLineType getPaySlipLineTypeByLabel(String label);
}

File: client/src/app/modules/accounting/services/pay.slip.line.type.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { PaySlipLineType } from '../../accounting/model/PaySlipLineType';

@Injectable({
  providedIn: 'root'
})
export class PaySlipLineTypeService extends AppRestService<PaySlipLineType>{

  constructor(http: HttpClient) {
    super(http, "accounting");
  }

  getPaySlipLineTypes() {
    return this.getList(new HttpParams(), "pay-slip-line-types");
  }
  
   addOrUpdatePaySlipLineType(paySlipLineType: PaySlipLineType) {
    return this.addOrUpdate(new HttpParams(), "pay-slip-line-type", paySlipLineType, "Enregistré", "Erreur lors de l'enregistrement");
  }

}
